shared/login/register/code-page/index.render.native.js

```diff
@@ -8,222 +8,223 @@ import Container from '../../forms/container'
+          <Text type='Header' style={{marginBottom: 10}}>Type in text code</Text>
+          <Text type='Body'>Please run</Text>
+          <Text type='TerminalInline' backgroundMode='Terminal'>keybase device add</Text>
+          <Text type='Body'>in the terminal on your computer.</Text>
+        </Box>
+        {this.renderShowText()}
+        {this.renderSwitchButton(codePageModeShowCode, 'icon-phone-qr-code-48', 'Scan QR code instead')}
+      </Container>
     )
   }
 
-  renderScanner () {
+  renderEnterTextForDesktop () {
     return (
-      <Qr
-        scanning={true}
-        onBarCodeRead={code => this.props.qrScanned(code)}
-        style={{flex: 1}}
-        qrCode={this.props.qrCode}>
-
-        {isIOS &&
-          <Box style={{alignSelf: 'center', width: 200, height: 200}}>
-            <Box style={[styles.box, styles.boxEdge, {left: 0}]} />
-            <Box style={[styles.box, styles.boxEdge, {right: 0}]} />
-            <Box style={[styles.box, styles.boxCorner, {right: 0, top: 0}]} />
-            <Box style={[styles.box, styles.boxCorner, {left: 0, top: 0}]} />
-            <Box style={[styles.box, styles.boxCorner, {right: 0, bottom: 0}]} />
-            <Box style={[styles.box, styles.boxCorner, {left: 0, bottom: 0}]} />
-          </Box>}
-      </Qr>
+      <Container style={stylesContainer} onBack={this.props.onBack}>
+        <Box style={stylesIntro}>
+          <Text type='Header'>Type in text code</Text>
+        </Box>
+        {this.renderEnterText()}
+        {this.renderSwitchButton(codePageModeScanCode, 'icon-phone-qr-code-48', 'Scan QR code instead')}
+      </Container>
     )
   }
 
-  render () {
+  renderTextForMobile () {
     return (
       <Container style={stylesContainer} onBack={this.props.onBack}>
-        {this.renderIntro()}
-        {this.renderContent()}
-        {this.renderSwitch()}
+        <Box style={stylesIntro}>
+          <Text type='Header' style={{marginBottom: 10}}>Type text code</Text>
+          <Text type='Body'>In the Keybase App</Text>
+          <Text type='Body'>{'go to Devices > Add a new device.'}</Text>
+        </Box>
+        <TabBar underlined={true} style={{flex: 1}}>
+          <TabBarItem
+            selected={this.props.mode === codePageModeShowText}
+            label='Display Code'
+            onClick={() => { this.props.setCodePageMode(codePageModeShowText) }}>
+            {this.renderShowText()}
+          </TabBarItem>
+          <TabBarItem
+            label='Type Code'
+            selected={this.props.mode === codePageModeEnterText}
+            onClick={() => { this.props.setCodePageMode(codePageModeEnterText) }}>
+            {this.renderEnterText()}
+          </TabBarItem>
+        </TabBar>
+        {this.renderSwitchButton(codePageModeScanCode, 'icon-phone-qr-code-48', 'Scan QR code instead')}
       </Container>
     )
   }
+
+  render () {
+    let isMobile = this.props.otherDeviceRole === codePageDeviceRoleExistingPhone
```

Comment:
It's whether the other device being used for provisioning is mobile. (Whether to show the tabbed views or not.)